cardano-node/src/Cardano/Tracing/Metrics.hs

```diff
@@ -79,29 +78,12 @@ instance HasKESMetricsData ByronBlock where
 
 instance All HasKESMetricsData xs => HasKESMetricsData (HardForkBlock xs) where
   getKESMetricsData _ forgeStateInfo =
-        combineKESMetrics
-      . hcollapse
-      . hcmap (Proxy @ HasKESMetricsData) getOne
-      . OptNP.toNP
-      . getPerEraForgeStateInfo
+        hcollapse
+      . hcmap (Proxy @HasKESMetricsData) getOne
+      . getOneEraForgeStateInfo
       $ forgeStateInfo
     where
       getOne :: forall blk. HasKESMetricsData blk
-             => (Maybe :.: WrapForgeStateInfo) blk
+             => WrapForgeStateInfo blk
              -> K KESMetricsData blk
-      getOne (Comp mbWrapForgeStateInfo) = K $
-          case unwrapForgeStateInfo <$> mbWrapForgeStateInfo of
-            Nothing -> NoKESMetricsData
-            Just oneForgeStateInfo ->
-              getKESMetricsData (Proxy @ blk) oneForgeStateInfo
-
-      -- Multiple eras can have 'KESMetricsData', e.g., Shelley and Shelley +
-      -- native assets. We don't know which era we're in, so we have to make a
-      -- choice: we choose the last era's 'KESMetricsData'.
-      --
-      -- TODO this might need revision when we have updated everything for the
-      -- first hard fork after Shelley, but before it actually happened. At
-      -- that moment, we'll trace the KESMetricsData of the upcoming era,
-      -- instead of the KESMetricsData of the current Shelley era.
-      combineKESMetrics :: [KESMetricsData] -> KESMetricsData
-      combineKESMetrics = lastDef NoKESMetricsData
```

Comment:
This PR solves the problem described in the TODO. For each Shelley-based era, the KES metrics will be traced independently. @deepfire this means that when a node is started with credentials for both Shelley and Allegra, that the two KES metrics will be traced per slot. Maybe they should be groupedd per label or something?

Reply:
@mrBliss, we don't particularly care for KES metrics in dense pools, I'd say -- so the less resources we spend here the better.

Reply:
Unlike the other two comments, this question is more about having two eras instead of having multiple stake pools.

> this means that when a node is started with credentials for both Shelley and Allegra, that the two KES metrics will be traced per slot.

Around the transition point from Shelley to Allegra, all block producing nodes should have credentials for both eras. I imagine the operators will want to see the metrics of both KES keys independently.

Reply:
Indeed, misunderstood you initially.

>  I imagine the operators will want to see the metrics of both KES keys independently.

Absolutely, yes.

Reply:
Right. As mentioned on Slack, this doesn't have to been done right away, it should just be done before the transition to the Allegra era (and any testnets that exercise such a transition). You can create an issue for it to not forget about it.

Reply:
@mrBliss, so in https://github.com/input-output-hk/cardano-node/tree/cad-1920-bulk-creds I've properly wired the credentials label into logs, so that label should now be displayed in structured logs.  I still need to test it though.

@mrBliss, now that I re-evaluate your question, it seems that tracking KES metrics separately per-epoch is fine -- they'll be labelled appropriately.

We'll need to accomodate it in RTView, though.